src/BesselFunctions/besselk.jl

```diff
@@ -614,3 +602,66 @@ end
         end
     )
 end
+
+# This is a version of Temme's proposed f_0 (1975 JCP, see reference above) that
+# swaps in a bunch of local expansions for functions that are well-behaved but
+# whose standard forms can't be naively evaluated by a computer at the origin.
+@inline function f0_local_expansion_v0(v, x)
+    l2dx = log(2/x)
+    mu   = v*l2dx
+    vv   = v*v
+    sp = evalpoly(vv, (1.0, 1.6449340668482264, 1.8940656589944918, 1.9711021825948702))
+    g1 = evalpoly(vv, (-0.5772156649015329, 0.04200263503409518, 0.042197734555544306))
+    g2 = evalpoly(vv, (1.0, -0.6558780715202539, 0.16653861138229145))
+    sh = evalpoly(mu*mu, (1.0, 0.16666666666666666, 0.008333333333333333, 0.0001984126984126984, 2.7557319223985893e-6))
```

Comment:
I'm kinda wondering how many terms we need for this expansion as mu slowly grows...
```julia
julia> v = 1e-4
0.0001

julia> v * log(2 / 24.0)
-0.0002484906649788
```
Five terms is probably ok ? I did some quick checks adding another term below but didn't seem to change much. Seems like a reasonable approximation that we have here. Just checking this `f0` is pretty accurate? This isn't contributing to errors we are seeing?
```
SeriesData[x, 0, {
 1.`20., 0, 0.16666666666666666666666666666666666667`20., 0, 
  0.00833333333333333333333333333333333333`20., 0, 
  0.00019841269841269841269841269841269841`20., 0, 
  2.75573192239858906525573192239859`20.*^-6, 0, 
  2.505210838544171877505210838544`20.*^-8}, 0, 11, 1]
```

Reply:
well those are huge `x` values---weren't we only going to use this for `x<1.5` or so?

In general, I'm sure more terms couldn't hurt---I can tinker with that. But it isn't obvious that it will help, because if `abs(v)<1e-5` and these are polynomials in `v^2`, then the sixth order coefficient will be at most `1e-30`. I know that the story is more complicated for derivatives though, so I'll see about putting in an extra term to see if it helps.

Reply:
Ya that's right haha. I will need to adjust cutoffs and stuff accordingly. So like what we probably should do is just have general routines for abs(v) < 1.5 and then use forward reccurence. It's much faster actually to do 2 Levin calculations and forward recurrence than it is to do the int power series. So let's adjust the whole routine. For v > 25 we will use the uniform expansion so forward recurrence will be fast here. 

But ya I think as long as we verify derivatives for v near 0.0 and x < 1.5 for this power series we should be ok. The tough thing about this though is that the derivatives are zero for v so something really close to v will be tricky to get right....

Reply:
Dang sorry ya I need to rethink the whole routine now that the intemediate range using the Levin transform is so fast. I used to just try to avoid that completely by extending the power series range as much as possible but now it makes sense to favor that when necessary and just fall back to forward recurrence when necessary.

It actually makes checking the accuracy of the whole routine much easier because we essentially have to just check our scalar and derivative information for v < 1.0 and we know that forward recurrence is also stable and accurate for besselk. This should greatly reduce the number of points we need to explicity check. Of course we should still do some scattered spotchecks to make sure the derivatives are carried out ok with AD